### DISCUSSION TITLE
Disable fetching release notes from certain domains

### DISCUSSION THREAD

**Discussion by Keysox** [renovatebot/renovate#14856] [Request Help]
Due to the way we're using the self-hosted instance of Renovate, github.com is not accessible. We would like the ability to turn off these warnings and only fetch from the hosts that we have access to. Is this something that's already supported or should I create an issue for it?

We don't want to disable this globally because we wouldn't get release notes for things we do have access to such as private dependencies.

For additional context, here's what we're currently seeing:
* Logs and dependency dashboard: `WARN: No github.com token has been configured. Skipping release notes retrieval`
* Pull requests: `Release Notes retrieval for this PR were skipped because no github.com credentials were available.`

**Comment by rarkins** [comment]
Do I understand this right?

- Renovate detects no github.com token, and correctly does not attempt to fetch release notes from github.com?
- However, it's disabling *all* release notes fetching while you would still like to fetch from your own host(s)?

**Reply by Keysox** [reply]
- Correct but we get warnings on every pull request, etc. leading to confusion/noise
- Release notes are working properly from our own host. However, I found https://docs.renovatebot.com/configuration-options/#fetchreleasenotes that would disable them globally, which is not quite what we want. It would be great to be able to suppress the warnings only from github.com since it's expected behavior to not fetch from there as it will never work due to a firewall that's in place

**Reply by rarkins** [reply]
Can you screenshot the warning you get on every Pull Request? If that warning disappeared, would everything be resolved for you?

**Reply by Keysox** [reply]
Sure thing! Here are some screenshots of where it occurs. And yes, that would be amazing :)

Pull Requests:
<img width="660" alt="Screen Shot 2022-03-30 at 9 49 22 AM" src="https://user-images.githubusercontent.com/6796642/160877040-47df5225-ff0b-438f-9358-70ae595f08cf.png">

Dependency Dashboard:
<img width="657" alt="Screen Shot 2022-03-30 at 9 49 45 AM" src="https://user-images.githubusercontent.com/6796642/160877128-52f9d5d4-50ae-4092-bc5e-fb5261ff8aab.png">

Logs:
<img width="582" alt="Screen Shot 2022-03-30 at 9 52 28 AM" src="https://user-images.githubusercontent.com/6796642/160877716-326fbe27-f5ea-4e22-bb93-bd8c0a6e1c74.png">

**Reply by rarkins** [reply]
I think we should introduce a new global config option to disable these warnings.

**Reply by Keysox** [reply]
I think that makes sense if we can limit it to disabling just release notes warnings or this token not being defined so others aren't hidden accidentally! Unsure if they should still be there in debug logs. Thanks

**Reply by rarkins** [reply]
Can you submit a feature request issue? Please copy in the screenshots you added here.

I suggest we can call the new option "githubTokenWarn" which defaults to true. If set to false (global config only) then ew no longer log a warning about it (which will make it disappear from the Dependency Dashboard to) nor should we add the warning text in PR bodies.

**Reply by Keysox** [reply]
Sure thing! https://github.com/renovatebot/renovate/issues/14882
